webdriver/protocol.py

```diff
@@ -0,0 +1,130 @@
+import contextlib
+import httplib
+import json
+import pytest
+import types
+
+import webdriver
+
+
+class HTTPRequest(object):
+    def __init__(self, host, port):
+        self.host = host
+        self.port = port
+
+    def head(self, path):
+        return self._request("HEAD", path)
+
+    def get(self, path):
+        return self._request("GET", path)
+
+    @contextlib.contextmanager
+    def _request(self, method, path):
+        conn = httplib.HTTPConnection(self.host, self.port)
+        try:
+            conn.request(method, path)
+            yield conn.getresponse()
+        finally:
+            conn.close()
+
+
+@pytest.fixture
+def http(request, session):
+    return HTTPRequest(session.transport.host, session.transport.port)
+
+
+@pytest.fixture
+def new_session(session):
+    session.end()
+    assert session.session_id is None
+    return session
+
+
+def test_provides_http(http):
+    # 4.0
+    with http.head("/") as resp:
+        assert resp.status == 404
```

Comment:
Seems fragile. Surely any http response is good? For example, an intermediary node could offer a nice UI for figuring out which sessions are active at a particular time at the root of the HTTP server and still be fully compliant with the spec.